### PR TITLE
more unit tests for code in gbits.c, gridtemplates, drstemplates, pdstemplates, plus doxygen improvements and code cleanup

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #8.

More unit tests for the functions in gbits.c and gridtemplates.c. Also documentation improvements.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
OK, I will stop adding testing on this branch, and start a new one. THis branch takes testing up to ~20% coverage:
image

File: src/g2_create.c [hunk 1]
```diff
@@ -1,43 +1,54 @@
...
+ * - listsec1(3) GRIB Local Tables Version Number ([Table 1.1]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-1.shtml)).
+ * - listsec1(4) Significance of Reference Time ([Table 1.2]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-2.shtml))
  * - listsec1[5] Reference Time - Year (4 digits)
  * - listsec1[6] Reference Time - Month
  * - listsec1[7] Reference Time - Day
  * - listsec1[8] Reference Time - Hour
  * - listsec1[9] Reference Time - Minute
  * - listsec1[10] Reference Time - Second
```
Comment (BrianCurtis-NOAA) [inline src/g2_create.c:47]: parentheses for consistency

Comment (edwardhartnett) [reply src/g2_create.c:47]: No, parens for Fortran, square brackets for C.

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: src/g2_create.c (hunk 1 above)
Comment (edwardhartnett) [reply src/g2_create.c:47]: I changed them all to square brackets in this file.

File: src/g2_create.c [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
+ * A call to g2_gribend() is required to complete GRIB2 message after
+ * all fields have been added.
  *
  * @param[in] cgrib Character array to contain the GRIB2 message. Must
  * be allocated large enough to store the entire GRIB2 message.
  * @param[in] listsec0 Contains information needed for GRIB Indicator
  * Section 0. Must be dimensioned >= 2.
- * - listsec0[0]=Discipline-GRIB Master Table Number (see Code Table 0.0)
- * - listsec0[1]=GRIB Edition Number (currently 2)
+ * - listsec0(0) Discipline-GRIB Master Table Number ([Code Table 0.0]
```
Comment (BrianCurtis-NOAA) [inline src/g2_create.c:27]: Square bracket

File: src/g2_create.c [hunk 3]
```diff
@@ -1,43 +1,54 @@
...
+ * - listsec1(0) Id of orginating centre ([Table 0]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/on388/table0.html)).
+ * - listsec1(1) Id of orginating sub-centre ([Table C]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/on388/tablec.html)).
+ * - listsec1(2) GRIB Master Tables Version Number ([Table 1.0]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-0.shtml)).
+ * - listsec1(3) GRIB Local Tables Version Number ([Table 1.1]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-1.shtml)).
+ * - listsec1(4) Significance of Reference Time ([Table 1.2]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-2.shtml))
```
Comment (BrianCurtis-NOAA) [inline src/g2_create.c:41]: Square brackets

File: src/g2_create.c [hunk 4]
```diff
@@ -1,43 +1,54 @@
...
  * - listsec1[7] Reference Time - Day
  * - listsec1[8] Reference Time - Hour
  * - listsec1[9] Reference Time - Minute
  * - listsec1[10] Reference Time - Second
- * - listsec1[11] Production status of data (Code Table 1.3)
- * - listsec1[12] Type of processed data (Code Table 1.4)
+ * - listsec1(11) Production status of data ([Table 1.3]
+ * (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-3.shtml)).
+ * - listsec1(12) Type of processed data ([Table 1.4]
+ *  (https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table1-4.shtml)).
```
Comment (BrianCurtis-NOAA) [inline src/g2_create.c:51]: Square brackets

File: src/g2_create.c (hunk 2 above)
Comment (edwardhartnett) [reply src/g2_create.c:27]: Fixed in next PR...

File: src/g2_create.c (hunk 3 above)
Comment (edwardhartnett) [reply src/g2_create.c:41]: Fixed in next PR...

File: src/g2_create.c (hunk 4 above)
Comment (edwardhartnett) [reply src/g2_create.c:51]: Fixed in next PR...

**Pull request merged**